### ISSUE TITLE
agent: watcher::tests::watch_directory_too_large sometimes panicked at 'expected error', src/watcher.rs:749:22

### ISSUE THREAD

**Issue by yibozhuang** [kata-containers/kata-containers#2640]
# Description of problem

In `src/agent/src/watcher.rs` we have a test that covers
that case where we expect to error when the number of files have exceeded `MAX_ENTRIES_PER_STORAGE` (which is 16)

https://github.com/kata-containers/kata-containers/blob/main/src/agent/src/watcher.rs#L680

The test creates 16 files in the source directory and expects `scan()` to return success with 16 files changed.
It then creates the 17th file in the directory and expects `scan()` to return `WatcherError::MountTooManyFiles`

In our internal CI testing, we sometimes see that this test fails with
```
---- watcher::tests::watch_directory_too_large stdout ----
     thread 'watcher::tests::watch_directory_too_large' panicked at 'expected error', src/watcher.rs:749:22
```

# Expected result

Expect this test to consistently pass and `scan()` call on creation of the 17th file should return `WatcherError::MountTooManyFiles`

# Actual result

Sometimes the test panic with unexpected error because `scan()` returns `Ok()`
https://github.com/kata-containers/kata-containers/blob/main/src/agent/src/watcher.rs#L749 

# Further information

(replace this text with any extra information you think might be useful)

**Comment by yibozhuang** [comment]
I intend to work on this issue and have a fix for this.

**Comment by yibozhuang** [comment]
Addressed by #2565. Closing.
